test/e2e/virtual/apiexport/virtualworkspace_test.go

```diff
@@ -178,6 +162,129 @@ func TestAPIExportVirtualWorkspace(t *testing.T) {
+
+	wildwestClusterClient, err := wildwestclientset.NewClusterForConfig(cfg)
+	require.NoError(t, err, "failed to construct wildwest cluster client for server")
+
+	apifixtures.CreateSheriffsSchemaAndExport(ctx, t, serviceProviderSherriffs, kcpClients, "wild.wild.west", "board the wanderer")
+
+	t.Logf("get the sheriffs api export's generated identity hash")
+	identityHash := ""
+	wait.PollImmediateWithContext(ctx, 100*time.Millisecond, wait.ForeverTestTimeout, func(c context.Context) (done bool, err error) {
+		sheriffExport, err := kcpClients.Cluster(serviceProviderSherriffs).ApisV1alpha1().APIExports().Get(ctx, "wild.wild.west", metav1.GetOptions{})
+		if err != nil {
+			return false, nil
+		}
+
+		if conditions.IsTrue(sheriffExport, apisv1alpha1.APIExportIdentityValid) {
+			identityHash = sheriffExport.Status.IdentityHash
+			return true, nil
+		}
+		return false, nil
+	})
+
+	t.Logf("Found identity hash: %v", identityHash)
+	apifixtures.BindToExport(ctx, t, serviceProviderSherriffs, "wild.wild.west", consumerWorkspace, kcpClients)
+	setUpServiceProviderWithPermissionClaims(ctx, dynamicClients, kcpClients, kubeClusterClient, serviceProviderWorkspace, t, identityHash)
+
+	bindConsumerToProvider(ctx, consumerWorkspace, serviceProviderWorkspace, t, kcpClients)
+
+	createCowboyInConsumer(ctx, t, consumerWorkspace, wildwestClusterClient)
+	t.Logf("test that the admin user can use the virtual workspace to get cowboys")
+	apiExport, err := kcpClients.Cluster(serviceProviderWorkspace).ApisV1alpha1().APIExports().Get(ctx, "today-cowboys", metav1.GetOptions{})
+	require.NoError(t, err, "error getting APIExport")
+	require.Len(t, apiExport.Status.VirtualWorkspaces, 1, "unexpected virtual workspace URLs: %#v", apiExport.Status.VirtualWorkspaces)
+
+	apiExportVWCfg := rest.CopyConfig(cfg)
+	apiExportVWCfg.Host = apiExport.Status.VirtualWorkspaces[0].URL
+
+	wildwestVCClients, err := wildwestclientset.NewClusterForConfig(apiExportVWCfg)
+	require.NoError(t, err)
+	cowboysProjected, err := wildwestVCClients.Cluster(logicalcluster.Wildcard).WildwestV1alpha1().Cowboys("").List(ctx, metav1.ListOptions{})
+	require.NoError(t, err)
+	require.Equal(t, 1, len(cowboysProjected.Items))
+
+	t.Logf("Ensuring the appropriate core/v1 resources are available")
+	dynamicVWClients, err := dynamic.NewClusterForConfig(apiExportVWCfg)
+	require.NoError(t, err, "error creating dynamic cluster client for %q", apiExportVWCfg.Host)
+
+	grantedGVRs := []schema.GroupVersionResource{
+		{Version: "v1", Resource: "namespaces"},
+		{Version: "v1", Resource: "configmaps"},
+		{Version: "v1", Resource: "secrets"},
+		{Version: "v1", Resource: "serviceaccounts"},
+		{Version: "v1", Resource: "sheriffs", Group: "wild.wild.west"},
+	}
+
+	for _, gvr := range grantedGVRs {
+		t.Logf("Trying to wildcard list %q", gvr)
+		_, err = dynamicVWClients.Cluster(logicalcluster.Wildcard).Resource(gvr).List(ctx, metav1.ListOptions{})
+		require.NoError(t, err, "error listing %q", gvr)
```

Comment:
Might want to add another export+consumer+sheriff combo to prove that the claim only pulls in instances from the correct identity hash (and add more checks to verify it's the correct one)

Reply:
Adding extra tests.